### ISSUE TITLE
Ability to log out with a http request

### ISSUE THREAD

**Issue by emedbo** [IdentityServer/IdentityServer4#481]
When you log in from an app to get a code (connect/authorize?response_type=code&response_mode=query&client_id=app&...&...) you need to add an authentication cookie in order to authenticate the user (like all the samples show). The app only fires up a web view to log the user in. I am using an external OpenIDConnect provider to handle user IDs/Passwords, and IdentityServer4 to handle tokens etc. 

In order to properly sign out, the app needs to open a web view again to call my /ui/logout controller, which is not acceptable. 

It seems not possible to use the /connect/endsession endpoint since the user will not be signed in using this endpoint as far as I have understood. 

So the app only stores id_token, access_token and refresh_token, and does not want to use the web view to logout. 

Is this not yet possible or have I misunderstood?
PS: I have debugged through the /connect/endsession endpoint

**Comment by Lutando** [comment]
I hope I understood your question correctly. But you can change what the controller method returns when you logout. In all the quickstart samples the controller methods return a view. You can return a redirect back to where the client requested on its endsession request.

So eg 

https://github.com/IdentityServer/IdentityServer4.Samples/blob/dev/Quickstarts/4_ImplicitFlowAuthenticationWithExternal/src/QuickstartIdentityServer/Controllers/AccountController.cs#L230

You could change to `return Redirect(vm.PostLogoutRedirectUri)` and there will be no web view saying you've logged out.

**Comment by brockallen** [comment]
> In order to properly sign out, the app needs to open a web view again to call my /ui/logout controller, which is not acceptable.

What does it mean to signout of a native app? If your webview does not maintain cookie state, then there's not need to pop a web view, is there? You can always use the revocation endpoint to revoke access or refresh tokens, so perhaps that's sufficient for what you need to do.

**Comment by emedbo** [comment]
Im not the expert for this front-end app, but they can't seem to delete cookies in the web view. I managed to do as @Lutando suggested. 
The thing is that we're using Azure B2C as user account provider, and they set a cookie. This means the next time you get redirected, you're already signed in, then get redirected back, unable to change user. 

Therefore we need to sign the user out. I was able to make a smooth sign-out that redirects back after clearing cookies. 

Another question: would this issue somehow solve my question? https://github.com/IdentityServer/IdentityServer4/issues/455
